### ISSUE TITLE
Null pointer exception while reading properties which are not part of request

### ISSUE THREAD

**Issue by buddha1990** [OfficeDev/ews-java-api#361]
Hello,
 We are getting NULL pointer exception while reading properties which are not part of request.
 The issue here is, while making request if i'm not setting some property like 'conference type' and later after creating appointment when I try to get that property value by getter, it will throw us NULL pointer
exception. As all those properties are not sending back when i'm loading the appointment ( appointment.load()).
This is causing issue while consuming the EWS (2.0 SNAPSHOT) APIs. In our code, we are reading all the properties, if any property is missing, exception is happening.
 Please provide any alternate approach to read the values.

**Comment by elewyeen** [comment]
So try:       
                  CalendarView view = new CalendarView(startDate, endDate);
			List<PropertyDefinition> params = new ArrayList<PropertyDefinition>();
			params.add(AppointmentSchema.Subject);
                         // You need all of the attributes    
			params.add(AppointmentSchema.ConferenceType);
			PropertyDefinition[] property = new PropertyDefinition[params
					.size()];
			params.toArray(property);
			FindItemsResults<Appointment> appointments = calendar.findAppointments(view);
		    service.loadPropertiesForItems((Iterable)appointments, new PropertySet(property));
                   for (Appointment appointment : appointments.getItems())
		    {
                        appointment.getConferenceType();
                     }

**Comment by vasugupta09** [comment]
Hi,

I tried the above approach, but it still giving the same error. I tried to use above code after appointment.save(); as using it before doesn't make sense. Can u give a specific example assuming that i have already created a meeting without sending conferenceType to the server, and now i have a response object by the server. 

Thanks

**Comment by elewyeen** [comment]
hi,
My exchange version is 2007 not 2007_sp1, add the following code:
service.setExchange2007CompatibilityMode(true);
save(),update(),cancel() All need to do so



zn85600301
 
From: vasugupta09
Date: 2015-07-13 14:31
To: OfficeDev/ews-java-api
CC: elewyeen
Subject: Re: [ews-java-api] Null pointer exception while reading properties which are not part of request (#361)
Hi,
I tried the above approach, but it still giving the same error. I tried to use above code after appointment.save(); as using it before doesn't make sense. Can u give a specific example assuming that i have already created a meeting without sending conferenceType to the server, and now i have a response object by the server. 
Thanks
